### DISCUSSION TITLE
Install my own local library

### DISCUSSION THREAD

**Discussion by v-danh** [python-poetry/poetry#9741] [Q&A]
This is the first time I have tried to write my library using `poetry` ., then I used `poetry build` and `poetry add` locally as follows:
`poetry build` is successful, but 
`poetry add` is not.

This is my dictionary structure tree

> ![Screenshot 2024-10-09 090745](https://github.com/user-attachments/assets/16a375e4-1df5-4e3e-b53b-ef9a3c1238ce)

> ![Screenshot 2024-10-09 085237](https://github.com/user-attachments/assets/5c80a8f9-636e-432e-bfca-9922e366f90e)

> Cannot add dependency on colorxcode to project with the same name.
No changes were applied.
![Screenshot 2024-10-09 085426](https://github.com/user-attachments/assets/95a773f0-cc4d-4d9d-9ddb-e54f4959efdf)

As far as I know in poetry's doc, 

> The package name should not be the same as the project name.

And why is it not allowed to have duplicate names?

**Comment by v-danh** [comment]
@Secrus or someone else explain to me in #9741?

**Comment by nbaju1** [comment]
Are you trying to add the package as a dependency to itself? If so, why?

> The package name should not be the same as the project name.

Where in the docs is this stated?
